src/reflect/type.go

```diff
@@ -487,7 +488,40 @@ func pointerTo(t *rawType) *rawType {
 }
 
 func (t *rawType) String() string {
-	return "T"
+	if t.isNamed() {
+		// TODO(dgryski): `main` until pkgPath support lands
+		return "main" + "." + t.Name()
```

Comment:
I would have called this something like `"<unknown>"` instead of `"main"` but as this is going to change very soon anyway it's no big deal.